src/components/container/SurveyTaskList/SurveyTaskList.tsx

```diff
@@ -28,19 +29,30 @@ export type SurveyTaskListListPreviewState = "IncompleteTasks" | "CompleteTasks"
 
 export default function (props: SurveyTaskListProps) {
 	const [loading, setLoading] = useState(true);
-	const [tasks, setTasks] = useState<SurveyTask[] | null>(null);
+	const [tasks, setTasksInner] = useState<SurveyTask[] | null>(null);
+	const [activeSurveys, setActiveSurveys] = useState<string[]>([]);
 	const context = useContext(LayoutContext);
 
-	useEffect(() => {
-		initialize()
-		MyDataHelps.on("applicationDidBecomeVisible", initialize);
-		return () => {
-			MyDataHelps.off("applicationDidBecomeVisible", initialize);
+	const setTasks = (tasks: SurveyTask[]): void => {
+		setTasksInner(tasks);
+		setActiveSurveys([]);
+	};
+
+	useInitializeView(initialize, [], [props.previewState]);
+
+	const isSurveyActive = (task: SurveyTask): boolean => {
+		return activeSurveys.includes(task.surveyName);
+	};
+
+	const onTaskClicked = (task: SurveyTask) => {
+		if (!loading && !isSurveyActive(task)) {
```

Comment:
I am not sure that the check for "loading" does anything actually, does it?  Was probably something I erroneously added in the first version of this.  There would be no tasks if this was loading so nothing to click...

Reply:
I removed the `!loading` check from the task click handler since `loading` will only be `true` when there are no tasks yet.